sdk/core/core-rest-pipeline/documentation/pipeline.md

```diff
@@ -0,0 +1,220 @@
+   * Options for how redirect responses are handled.
+   */
+  redirectOptions?: RedirectPolicyOptions;
+
+  /**
+   * Options for adding user agent details to outgoing requests.
+   */
+  userAgentOptions?: UserAgentPolicyOptions;
+}
+
+/**
+ * Defines options that are used to configure internal options of
+ * the HTTP pipeline for an SDK client.
+ */
+export interface InternalPipelineOptions extends PipelineOptions {
+  /**
+   * Options to configure request/response logging.
+   */
+  loggingOptions?: LogPolicyOptions;
+}
+
+/**
+ * Create a new pipeline with a default set of customizable policies.
+ * @param options - Options to configure a custom pipeline.
+ */
+declare function createPipelineFromOptions(options: InternalPipelineOptions): Pipeline;
+```
+
+### Policies included in a default pipeline
+
+- proxyPolicy **(NODE Only)**
+  - A policy that allows one to apply proxy settings to all requests. If not passed static settings, they will be retrieved from the HTTPS_PROXY or HTTP_PROXY environment variables.
+- decompressResponsePolicy **(NODE Only)**
+  - A policy to enable response decompression according to Accept-Encoding header
+- formDataPolicy
+  - A policy that encodes FormData on the request into the body.
+- userAgentPolicy
+  - A policy that sets the User-Agent header (or equivalent) to reflect the library version
+- setClientRequestIdPolicy
+  - The programmatic identifier of the setClientRequestIdPolicy
+- defaultRetryPolicy
+  - A policy that retries according to three strategies:
+    - When the server sends a 429 response with a Retry-After header.
+    - When there are errors in the underlying transport layer (e.g. DNS lookup failures).
+    - Or otherwise if the outgoing request fails, it will retry with an exponentially increasing delay.
+- tracingPolicy
+  - A simple policy to create OpenTelemetry Spans for each request made by the pipeline that has SpanOptions with a parent. Requests made without a parent Span will not be recorded.
+- redirectPolicy
+  - A policy to follow Location headers from the server in order to support server-side redirection.
+- logPolicy
+  - A policy that logs all requests and responses.
+
+There are other policies commonly added to the default pipeline:
+
+- bearerTokenAuthenticationPolicy
+  - A policy that can request a token from a TokenCredential implementation and then apply it to the Authorization header of a request as a Bearer token.
+- serializationPolicy
+  - This policy handles assembling the request body and headers using an OperationSpec and OperationArguments on the request.
+- deserializationPolicy
+  - This policy handles parsing out responses according to OperationSpecs on the request.
+
+# Configuring client pipeline
+
+When instantiating a new client a pipeline can be configured by adding additional policies to the default pipeline, passing a static array of policies.
+
+```typescript
+const sendRequest = (request: PipelineRequest, next: SendRequest) => next(request);
```

Comment:
this type is defined by core-rest-pipeline publicly as `SendRequest` if you'd like to import it for the snippet